flexget/plugins/sites/torrentleech.py

```diff
@@ -121,48 +120,35 @@ def search(self, task, entry, config=None):
         for search_string in entry.get('search_strings', [entry['title']]):
             query = normalize_unicode(search_string).replace(":", "")
-            # urllib.quote will crash if the unicode string has non ascii characters, so encode in utf-8 beforehand
-            url = ('https://v4.torrentleech.org/torrents/browse/index/query/' +
+            # urllib.quote will crash if the unicode string has non ascii characters,
+            # so encode in utf-8 beforehand
+
+            url = ('https://www.torrentleech.org/torrents/browse/list/query/' +
                    quote(query.encode('utf-8')) + filter_url)
-            log.debug('Using %s as torrentleech search url' % url)
+            log.debug('Using {} as torrentleech search url'.format(url))
 
-            page = task.requests.get(url, cookies=login.cookies).content
-            soup = get_soup(page)
+            results = task.requests.get(url, cookies=login.cookies).json()
 
-            for tr in soup.find_all("tr", ["even", "odd"]):
-                # within each even or odd row, find the torrent names
-                link = tr.find("a", attrs={'href': re.compile('/torrent/\d+')})
-                log.debug('link phase: %s' % link.contents[0])
+            for torrent in results['torrentList']:
                 entry = Entry()
-                # extracts the contents of the <a>titlename/<a> tag
-                entry['title'] = link.contents[0]
-
-                # find download link
-                torrent_url = tr.find("a", attrs={'href': re.compile('/download/\d+/.*')}).get('href')
-                # parse link and split along /download/12345 and /name.torrent
-                download_url = re.search('(/download/\d+)/(.+\.torrent)', torrent_url)
-                # change link to rss and splice in rss_key
-                torrent_url = 'https://v4.torrentleech.org/rss' + download_url.group(1) + '/' \
-                              + rss_key + '/' + download_url.group(2)
-                log.debug('RSS-ified download link: %s' % torrent_url)
-                entry['url'] = torrent_url
-
-                # us tr object for seeders/leechers
-                seeders, leechers = tr.find_all('td', ["seeders", "leechers"])
-                entry['torrent_seeds'] = int(seeders.contents[0])
-                entry['torrent_leeches'] = int(leechers.contents[0])
-                entry['search_sort'] = torrent_availability(entry['torrent_seeds'], entry['torrent_leeches'])
+                entry['title'] = torrent['name']
 
-                # use tr object for size
-                size = tr.find("td", text=re.compile('([\.\d]+) ([TGMK]?)B')).contents[0]
-                size = re.search('([\.\d]+) ([TGMK]?)B', size)
-
-                entry['content_size'] = parse_filesize(size.group(0))
+                # construct download URL
+                torrent_url = 'https://www.torrentleech.org/rss/download/{}/{}/{}'.format(
+                    torrent['fid'], rss_key, torrent['filename'])
```

Comment:
Is the linebreak necessary? We allow up to 120 chars on one line.

Reply:
I will adjust my IDE accordingly :)

Reply:
This particular line is over 120 chars too though.

Reply:
This particular line is still above 120 chars though.